client/components/Widget.jsx

```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import styled from 'styled-components';
 // import React from 'react-dom';
 
 class Widget extends React.Component {
```

Comment:
Perhaps refactor this into a functional stateless component? I see that you set a state variable from props. I read that that is generally considered an anti-pattern. I was having issues with this in connect 4 myself. Doesn't look like you use this state variable anyways.

Reply:
Maybe you're right.  there's going to be so much logic involving the calendar picker and I was considering storing selected dates in this component for separation and testing purposes.  Is it considered an anti pattern just due to redundancies or creating an extra layer to the data flow?  If I don't end up refactoring to a functional component then maybe I can alter that.

Reply:
It has to do with creating an extra layer and because it will not update upon change I belive. i.e. passing  a new listingId won't reset the state variable for this component. Passing it down as props would probably get the job done? This article does a better job of explaining it lol. https://codeburst.io/how-to-not-react-common-anti-patterns-and-gotchas-in-react-40141fe0dcd

Reply:
> It has to do with creating an extra layer and because it will not update upon change I belive. i.e. passing a new listingId won't reset the state variable for this component. Passing it down as props would probably get the job done? This article does a better job of explaining it lol. https://codeburst.io/how-to-not-react-common-anti-patterns-and-gotchas-in-react-40141fe0dcd

Passing the listingid as props would update i mean.